### PR TITLE
New deprecated annotations

### PR DESCRIPTION
**Pull request by autoantwort**
Depends on https://github.com/microsoft/vcpkg/pull/29777

### CODE REVIEW

File: .github/workflows/untrustedPR.yml [hunk 1]
```diff
@@ -34,10 +26,11 @@ jobs:
 
           unset VCPKG_ROOT
 
-          git diff --name-status --merge-base HEAD^ HEAD --diff-filter=MAR -- '*portfile.cmake' | sed 's/[MAR]\t*//' | while read filename; do grep -q -E '(vcpkg_apply_patches|vcpkg_copy_tool_dependencies|vcpkg_build_msbuild|vcpkg_extract_source_archive_ex|vcpkg_install_cmake|vcpkg_build_cmake|vcpkg_configure_cmake|vcpkg_fixup_cmake_targets)' "$filename" && echo " - \`$filename\`" || true; done > .github-pr.deprecated-cmake
+          git diff --name-status --merge-base HEAD^ HEAD --diff-filter=MAR -- '*portfile.cmake' | sed 's/[MAR]\t*//' > .github-pr.changed-portfiles
+          if [ -s .github-pr.changed-portfiles ]; then grep -n -H -E '(vcpkg_apply_patches|vcpkg_copy_tool_dependencies|vcpkg_build_msbuild|vcpkg_extract_source_archive_ex|vcpkg_install_cmake|vcpkg_build_cmake|vcpkg_configure_cmake|vcpkg_fixup_cmake_targets)' $(cat .github-pr.changed-portfiles) > .github-pr.deprecated-cmake; else touch .github-pr.deprecated-cmake; fi
```
Comment (dg0yt) [inline .github/workflows/untrustedPR.yml:30]: Can we either 
- generally disable the check for `vcpkg_copy_tool_dependencies`, or
- supress this check for ports which use `vcpkg_configure_make`, or
- add some flag to disable the check? 

There are false positive warnings when this is used for tools in `tools/${PORT}/bin` or `manual-tools/${PORT}`.

Comment (autoantwort) [reply .github/workflows/untrustedPR.yml:30]: Yeah I can remove the check for `vcpkg_copy_tool_dependencies`

Comment (autoantwort) [reply .github/workflows/untrustedPR.yml:30]: According to https://learn.microsoft.com/en-us/vcpkg/maintainers/functions/vcpkg_copy_tool_dependencies the function is also not deprecated anymore.

**Comment by Cheney-W** [comment]
@autoantwort Could you please solve the conflicts? Thank you!

**Review by BillyONeal — DISMISSED** [review]
This rocks <3

This approval is conditional on the intentional breakage being reverted of course :)

**Comment by autoantwort** [comment]
As reference 
image

**Review by BillyONeal — APPROVED** [review]

**Comment by Cheney-W** [comment]
@autoantwort Some PR blocked in the `Check For Common Mistakes` step, the `Formatting job` always return `Error: Process completed with exit code 1`, could you please take a look?

https://github.com/microsoft/vcpkg/actions/runs/4259936591/jobs/7412616549

**Pull request merged**